src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2891,9 +2891,12 @@ private static bool IsValidStatementExpression(SyntaxNode syntax, BoundExpressio
             // is not legal because it is a delegate-creation-expression and not an
             // object-creation-expression, but of course we don't know that syntactically.
 
-            if (expression.Kind == BoundKind.DelegateCreationExpression || expression.Kind == BoundKind.NameOfOperator)
+            switch (expression.Kind)
             {
-                return false;
+                case BoundKind.DelegateCreationExpression:
+                case BoundKind.NameOfOperator:
+                case BoundKind.UnboundObjectCreationExpression:
```

Comment:
Although we know that this can't be used as an statement expression from the syntax, I didn't modify `SyntaxFacts.IsStatementExpression` because the grammar allows it.